### PR TITLE
Search block styles in the style variations, option 1

### PR DESCRIPTION
**Pull request by carolinan**
**Description**

This PR changes the border radius of the search block button and search input field.
It also sets the border color of the search input field to accent color 6, the color with the opacity.
In one variation, it also updates the font family for the button.

With this option, the input field styles are limited to this single block, and managed in multiple theme.json files.

**Screenshots**
Noon, after

Search block styles in the Noon variation

Dusk, after:
Search block styles in the Dusk variation

**Testing Instructions**

Add a search form.
Confirm that the style matches the expectations listed in this issue: https://github.com/WordPress/twentytwentyfive/issues/485
Repeat the test for all style variations

### CODE REVIEW

**Comment by beafialho** [comment]
I'm having issues with my testing setup and I don't see the same as your screenshot 😞 However, if that's how it looks for you, it's looking as intended.

**Comment by carolinan** [comment]
Yes the shadow for example is working well for me on the search block button.

But I am concerned about how well the border radius on the input works with other inputs, mainly the comment form which  always has a border radius of 4px.

**Comment by carolinan** [comment]
Like in this screenshot
inputs

**Comment by beafialho** [comment]
> the comment form which always has a border radius of 4px.

Visually, the comment form radius should inherit the styles of each variation.

Also, if we could remove this option for the search block, it would be good. It looks less than ideal in every variation.

Captura de ecrã 2024-10-09, às 11 41 44

**Comment by carolinan** [comment]
There is no way for the theme to remove this option, that I am aware of 🤔 Not even with PHP or JavaScript.

**Comment by carolinan** [comment]
>Visually, the comment form radius should inherit the styles of each variation.

I agree with this. Short term I can try to add it to this PR.

I am still thinking of how we could do this for all inputs including the third party blocks as requested here:
https://github.com/WordPress/twentytwentyfive/issues/484. Yes, I opened this issue, but because it was requested by a user on Twitter.
In case you missed it, especially the select list looks bad in my opinion.

**Comment by beafialho** [comment]
Yes, I saw that. It would be great if we could do it for all inputs.

> In case you missed it, especially the select list looks bad in my opinion.

I have [replied with a visual suggestion](https://github.com/WordPress/twentytwentyfive/issues/484#issuecomment-2401747105) for the select list.

**Comment by carolinan** [comment]
If we handle the comment form and the select list in separate PR's, this is ready for review.

File: styles/08-midnight.json [hunk 1]
```diff
@@ -225,8 +225,13 @@
...
 			"core/search": {
-				"border": {
-					"radius": "0px"
+				"css": "& .wp-block-search__input{border-radius:0px;border-color:var(--wp--preset--color--accent-6);}",
+				"elements": {
+					"button": {
+						"border": {
+							"radius": "0px"
+						}
+					}
```
Comment (carolinan) [inline styles/08-midnight.json:234]: This is not valid, is it? These are not typography styles.

**Review by juanfra — COMMENTED** [review]
It looks well. I noticed that when you switch the style variation to one of these and then go back to one where the input has a rounded shape, then that doesn't take effect. For example, trying to go back to the default style.


https://github.com/user-attachments/assets/5bdd6d04-2a6d-4c7a-9f6a-ef445d663933

**Comment by carolinan** [comment]
How strange.
It looks like the variations that are supposed to have 0 border radius on the inputs still have 4px.

My guess would be that it is the styles> blocks > block name > CSS that is not clearing / refreshing.

**Comment by carolinan** [comment]
Because of this bug, I don't think it is worth implementing this change. I think it might be too confusing that the style changes depending on what order you preview the theme variations in...

**Comment by carolinan** [comment]
This still needs a decision.

**Comment by beafialho** [comment]
The default variation looks ok, as well as Noon and Dusk, but the other input fields aren't matching the buttons.

**Comment by carolinan** [comment]
How are you testing it?
Are you saving and resetting the style variation every time you are changing variation?

If you are not saving and resetting, then you are running into the the editor bug that Juan mentioned, and this is why I think this change should not be implemented, the bug is too confusing.

**Comment by beafialho** [comment]
I'm saving the variation, seeing on the front end, I go back to the editor and pick another variation.

**Comment by carolinan** [comment]
Yes, that's it. When we don't click on "reset styles" in between, the CSS for the input field is not updated, that is the bug.
I think this bug is blocking.

What we can do is open an issue for it on Trac, so that once it is fixed in the editor, the theme can be updated.
(There is already an issue for it in the Gutenberg GitHub repository)

**Pull request closed without merging**